### PR TITLE
Visitor can submit route

### PR DESCRIPTION
**Pull request by johanperjulius1**
## [PT](https://www.pivotaltracker.com/story/show/175524200)

- Input fields placeholder From & To
- Submit Button
- Using Google API to send a request to get the response with time and distance(km)

Skärmavbild 2020-11-04 kl  16 39 17

Skärmavbild 2020-11-04 kl  16 39 40

### CODE REVIEW

File: cypress/integration/visitorCanSubmitRoute.feature.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    cy.route({
+      method: "POST",
+      url:
+        `https://cors-anywhere.herokuapp.com/https://maps.googleapis.com/maps/api/directions/json?origin=Stockholm&destination=Orebro&key=${apiKey}`,
+      response: "fixture:stockholmOrebroResponse.json",
+    });
+    cy.visit("/");
+  });
+
+  it("visitor can see route form", () => {
```
Comment (oliverochman) [inline cypress/integration/visitorCanSubmitRoute.feature.js]: unnecessary test, please remove

File: cypress/integration/visitorCanSubmitRoute.feature.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+      cy.get("[data-cy='route-form']").within(() => {
+        cy.get("[data-cy='formOrigin']").type("Stockholm");
+        cy.get("[data-cy='formDestination']").type("Ankeborg");
+        cy.get("[data-cy='submit-route']").click();
+      });
+
+      cy.get("[data-cy='failure-message']").within(() => {
+        cy.get("[data-cy='fail-message']").should(
+          "contain",
+          "Cannot find location."
```
Comment (oliverochman) [inline cypress/integration/visitorCanSubmitRoute.feature.js]: Iw ould add a bit more user-friendly error message

File: src/App.jsx [hunk 3]
```diff
@@ -1,11 +1,12 @@
...
+import RouteForm from "./components/RouteForm";
 
 const App = () => {
   return (
-    <div>
-      
-    </div>
-  )
-}
+    <>
```
Comment (oliverochman) [inline src/App.jsx]: no need for fragments here since you only rendering one element / component

File: src/components/RouteForm.jsx [hunk 4]
```diff
@@ -0,0 +1,88 @@
+import React, { useState } from "react";
+import { Button, Form, Container, Message } from "semantic-ui-react";
+import axios from "axios";
+
+const RouteForm = () => {
+  const [routeInformation, setRouteInformation] = useState();
+  const [failureMessage, setFailureMessage] = useState("");
+  const postRoute = async (event) => {
```
Comment (oliverochman) [inline src/components/RouteForm.jsx]: extract this method to module

**Review by oliverochman — CHANGES_REQUESTED** [review]

**Review by grconnor — APPROVED** [review]
Happy with all changes and additions.

**Review by SebastianN97 — APPROVED** [review]

File: src/components/RouteForm.jsx [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+const RouteForm = () => {
+  const [routeInformation, setRouteInformation] = useState();
+  const [invalidLocationMessage, setInvalidLocationMessage] = useState("");
+
+  const createRoute = async (event) => {
+    event.preventDefault();
+
+    const from = event.target.origin.value;
+    const to = event.target.destination.value;
+    const response = await getRoute.create(from, to)
```
Comment (emtalen) [inline src/components/RouteForm.jsx]: getRoute.create? is it a get request or a post request? 🤔

File: src/modules/route_request.js [hunk 6]
```diff
@@ -0,0 +1,22 @@
+import axios from "axios";
+
+const apiKey = process.env.REACT_APP_MAPSDIRECTIONS_API_KEY;
+
+const getRoute = {
```
Comment (emtalen) [inline src/modules/route_request.js]: rename this to something more generic for all requests

File: src/components/RouteForm.jsx (hunk 5 above)
Comment (grconnor) [reply src/components/RouteForm.jsx]: it's a post request, we just changed the naming for getRoute to calculateRoute.

File: src/modules/route_request.js [hunk 7]
```diff
@@ -19,4 +19,4 @@ const getRoute = {
   },
 };
 
-export default getRoute;
+export default calculateRoute;
```
Comment (emtalen) [inline src/modules/route_request.js]: change this to `export { name }` and then also change the import where you have it to `import { name } from...` that is the standard with modules

**Pull request merged**